src/main/scala/org/broadinstitute/hail/asm4s/FunctionBuilder.scala

```diff
@@ -88,110 +88,402 @@ abstract class FunctionBuilder[R](parameterTypeInfo: Array[TypeInfo[_]], returnT
     f.put(null, rhs)
   }
 
-  def newArray[T](size: Code[Int])(implicit tti: TypeInfo[T], atti: TypeInfo[Array[T]]): Code[Array[T]] = {
-    new Code[Array[T]] {
-      def emit(il: InsnList): Unit = {
-        size.emit(il)
-        il.add(tti.newArray())
-      }
-    }
-  }
-
-  def newInstance[T](parameterTypes: Array[Class[_]], args: Array[Code[_]])(implicit tct: ClassTag[T], tti: TypeInfo[T]): Code[T] = {
-    new Code[T] {
-      def emit(il: InsnList): Unit = {
-        il.add(new TypeInsnNode(NEW, Type.getInternalName(tct.runtimeClass)))
-        il.add(new InsnNode(DUP))
-        Invokeable.lookupConstructor[T](Array()).invoke(null, Array()).emit(il)
-      }
-    }
-  }
-
-  def newInstance[T]()(implicit tct: ClassTag[T], tti: TypeInfo[T]): Code[T] =
-    newInstance[T](Array[Class[_]](), Array[Code[_]]())
-
-  def newInstance[T, A1](a1: Code[A1])(implicit a1ct: ClassTag[A1],
-    tct: ClassTag[T], tti: TypeInfo[T]): Code[T] =
-    newInstance[T](Array[Class[_]](a1ct.runtimeClass), Array[Code[_]](a1))
-
-  def newInstance[T, A1, A2](a1: Code[A1], a2: Code[A2])(implicit a1ct: ClassTag[A1], a2ct: ClassTag[A2],
-    tct: ClassTag[T], tti: TypeInfo[T]): Code[T] =
-    newInstance[T](Array[Class[_]](a1ct.runtimeClass, a2ct.runtimeClass), Array[Code[_]](a1, a2))
-
   def getArg[T](i: Int)(implicit tti: TypeInfo[T]): LocalRef[T] = {
     assert(i >= 0)
-    assert(i < parameterTypeInfo.length)
+    assert(i < parameterTypeInfo.length + 1)
     new LocalRef[T](argIndex(i))
   }
 
-  def resultClass(c: Code[R]): Class[_] = {
+  def classAsBytes(c: Code[R]): Array[Byte] = {
     mn.instructions.add(start)
-    c.emit(mn.instructions)
+    val l = new java.util.concurrent.ConcurrentLinkedQueue[AbstractInsnNode]()
+    c.emit(l)
+    val dupes = l.groupBy(x => x).filter(_._2.toArray.length > 1).toSeq
+    if (!dupes.isEmpty)
+      println(s"dupes: $dupes")
+    l.foreach { x =>
+      mn.instructions.add(x)
+    }
     mn.instructions.add(new InsnNode(returnTypeInfo.returnOp))
     mn.instructions.add(end)
 
+    {
+      // print bytecode for debugging
+      val cwNoMaxesNoFrames = new ClassWriter(0)
+      cn.accept(cwNoMaxesNoFrames)
+      val cr = new ClassReader(cwNoMaxesNoFrames.toByteArray())
+      val tcv = new TraceClassVisitor(null, new Textifier, new PrintWriter(System.out))
+      cr.accept(tcv, 0)
+
+      // val sw = new StringWriter()
+      // CheckClassAdapter.verify(cr, false, new PrintWriter(sw))
+      // if (sw.toString().length() != 0) {
+      //   println("Verify Output for " + name + ":")
+      //   try {
+      //     val out = new BufferedWriter(new FileWriter("ByteCodeOutput.txt"))
+      //     out.write(sw.toString())
+      //     out.close()
+      //   } catch {
+      //     case e: IOException => System.out.println("Exception " + e)
+      //   }
+      //   println(sw)
+      //   println("Bytecode failed verification 1")
```

Comment:
This block is useful for debugging when you've screwed up the stack royally, because it ignores the stack and shows you other, possibly more fundamental, problems.